### ISSUE TITLE
Correct version but "Someone you added does not support new groups and needs to update Signal"

### ISSUE THREAD

**Issue by alternativesurfer** [signalapp/Signal-Android#10090]
- [X] I have searched open and closed issues for duplicates
- [X] I am submitting a bug report for existing functionality that does not work as intended
- [X] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [X] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
I have a user who cannot be invited into Signal's "new groups" (with admin support).

Error:
"Someone you added does not support new groups and needs to update Signal"

I have confirmed they are running on the same version of Signal as is on my test iPhone.
iOS 12
Signal: 3.19.0.7

### Steps to reproduce
- Create new group
- Invite user
- Error: "Someone you added does not support new groups and needs to update Signal"

**Actual result:** User on the correct version of Signal is not added to group.
**Expected result:** User on the correct version of Signal is added to the new group.

### Screenshots
![signal-2020-10-15-102316](https://user-images.githubusercontent.com/10135929/96164826-a6d69800-0ed0-11eb-99e8-150f86591772.jpeg)

### Device info
Inviter device(s):
Pixel 3
Android: 11
Signal: 4.72.5

iPhone 6s
iOS 12
Signal: 3.19.0.7

Invitee Device:
iPhone 7
iOS 12
Signal: 3.19.0.7

### Link to debug log
https://debuglogs.org/a6e34b93ce4dbc7567c7865b13bba0124eae2df59ce25a6cf72ad39e68e13e0b

**Comment by Tursko** [comment]
This is probably happening because groups v2 is controlled by a feature flag. Give your invitee's device some more time.

**Comment by Nisc3d** [comment]
Also make sure if there are any linked devices that they are on the latest version too.

**Comment by offlinehoster** [comment]
We tested this feature yesterday evening.

2 Participants. I sent invites...the person who received the invite has the latest version 4.73.4 from the Google Store. Me too,

I can invite some other people with this version , but some others not.

We also turned off the desktop app, should we "close" the desktop app?  

I'm nit sure what exactly the problem might be, but it exists.

**Comment by joshbowyer** [comment]
I am also having the same issue. The contact upgraded to the latest version via the in-app update on the same day I did, and yet I am unable to add them to a New Group.

Is this "feature flag" somehow dependant on Google Play? Because myself and the contact installed via apk on LineageOS, but another contact who can be added to a New Group installed via Play Store.

**Comment by Tursko** [comment]
This will be solved on the next update since they're completely removing the flags.

**Comment by CodingAnarchy** [comment]
I'm having this problem inviting someone as well: all my iOS friends have been able to be invited, but many of the Android users are not, despite being on the latest version.

**Comment by greyson-signal** [comment]
If they are on the latest mobile clients, the problem is likely that they're on an outdated desktop client. If they have old desktops they're not using anymore, they can unlink them from the mobile app.

**Comment by dotnetCarpenter** [comment]
@greyson-signal; My friend deleted her account on Signal desktop and then uninstalled Signal Desktop. Got the latest android Signal version from Play store (but not latest release on GitHub).

Still, I can not add her to one group before she upgrades. But she got nothing to upgrade.

**Comment by jedie** [comment]
I have the same problem with a friend. https://debuglogs.org/a6eef590d478a36fa5a711e8838a741fbc055331bc457db2c9e4f62eb765a45b

**Comment by WildPenquin** [comment]
I'm having this problem with several contacts which are new to Signal (installed during the last few 48 hours) while with others, the feature works.

Moreover, I can not send any Signal messages to these users (only SMS, i.e. grey non-locked icon). It is as if the client discovery is not working at all with some users. Refreshing the list by composing a new message does not add these contacts to Signal.

**Comment by lywyn** [comment]
Same here, moved friends over but one friend on iOS4.3 and the latest app cannot be added to the group for the same reason. Has made it a nightmare trying to get off WhatsApp having to keep groups there for now

**Comment by virtadpt** [comment]
Just had the same problem.  I had a copy of signald linked to my account that wasn't on the latest and greatest.  I unlinked it and the problem seems to have fixed itself.

**Comment by antonok-edm** [comment]
Also experienced this issue. A group of 5 people was upgraded from legacy, two members were kicked, and only one could be re-invited after upgrading. The other kicked member was present in the list of members who would be kicked, but was already running Signal 5.4.7. When trying to re-invite that member individually, I got the "Someone you added does not support new groups and needs to update Signal" toast notification. I am also running Signal 5.4.7.

**Comment by drlight-code** [comment]
I also run into this, I can't be added to any of my former groups. I have to mention that I was using only the desktop client for several months and never opened the mobile client, I wonder if this could be related. After reading about this issue I unlinked all devices, upgraded to 5.4.7 on Android, cleared caches and retried, however the problem persists. Unfortunately this renders Signal basically unusable for me since several weeks.

**Comment by dpw700** [comment]
I had the same problem. Couldn't add a mate of mine to a group b/c he needs to update his version of signal.

He un-linked his PC's and it came good.

**Comment by greyson-signal** [comment]
It's really hard to track down much of what's happening here, since bugs have been fixed over time. Having old linked devices is definitely the biggest culprit, especially non-official stuff like signal-cli or signald.

My overall suggestion for people still having issues:
1. Unlink all devices (Settings > Linked Devices)
2. If that doesn't work, just block and then unblock someone. Anyone, doesn't matter. Blocking a person triggers a process that will coincidentally also update your capabilities with the service.

If things still don't work (or even if 1 doesn't work -- you really shouldn't have to do 2), please open a new issue. Thanks!

**Comment by asher-dev** [comment]
I know this is very old, but I've just been troubleshooting for a friend who had this message come up while adding someone to a group. Another group member was able to add the person when she could not.

The person who got the error had not previously interacted with the new group member, while the person who successfully added them had previously messaged them.

The user who got the error was using Signal for Android 6.22.8. (I don't have the information for the other users.)
